### ISSUE TITLE
More informative exceptions

### ISSUE THREAD

**Issue by netdown** [cmpayc/react-native-fs-turbo#4]
It would be great to throw the OS level exceptions instead of a generic message.

For example, when a file can't be read I receive the following:
`Error: Can not open file`

With `react-native-file-access`, I receive:
`java.io.FileNotFoundException: /storage/emulated/0/Pictures/IMG_20241128_174015.jpg: open failed: EACCES (Permission denied)`

With the latter, I will also immediately know whether the file actually exists or not. Also, these might end up as unhandled errors in a logger, so it's very useful to include as much info as possible.

**Comment by cmpayc** [comment]
@netdown Yes, I agree. I haven't worked much on the error output yet. I'll have to make them more informative.
